toolchain/sem_ir/formatter.cpp

```diff
@@ -386,11 +386,12 @@ class NodeNamer {
         nodes[node_id.index] = {scope_idx, scope.nodes.AllocateName(
                                                *this, node.parse_node(), name)};
       };
-      auto add_node_name_id = [&](StringId name_id) {
+      auto add_node_name_id = [&](StringId name_id,
+                                  llvm::StringRef suffix = "") {
         if (name_id.is_valid()) {
-          add_node_name(sem_ir_.strings().Get(name_id).str());
+          add_node_name((sem_ir_.strings().Get(name_id).str() + suffix).str());
         } else {
-          add_node_name("");
+          add_node_name(suffix.str());
```

Comment:
I'm trying to find this in case run in a test (where only a suffix is printed), can you point me in the right direction?

Reply:
I've tried to create such a test case, but I don't think the case where the name is invalid and a suffix is present is currently reachable -- we happen to never create the nodes in question with an invalid `NameId` -- so we'll always end up using the name `""` here.

Reply:
Is that changing, or expected behavior? Could a CHECK note the behavior?

Reply:
I would expect that we will have cases in the future where the name is invalid. Node naming is best-effort, so falling back to an empty string / just the suffix in the case where we don't find a good name seems reasonable to me. Also, generally I've tried to make the formatter be robust against finding invalid IDs even in places where we wouldn't usually expect them, so that the output can still be used for inspecting the result of possibly broken checker behavior.